pkg/controller/plan/kubevirt.go

```diff
@@ -958,24 +957,24 @@ func (r *KubeVirt) UpdateVmByConvertedConfig(vm *plan.VMStatus, pod *core.Pod, s
 	}
 	defer resp.Body.Close()
 
-	vmConfigBytes, err := io.ReadAll(resp.Body)
+	vmConf, err := io.ReadAll(resp.Body)
 	if err != nil {
 		err = liberr.Wrap(err)
 		return
 	}
-	vmConfigXML := string(vmConfigBytes)
-
 	switch r.Source.Provider.Type() {
 	case api.Ova:
-		if vm.Firmware, err = ovfparser.GetFirmwareFromConfig(vmConfigXML); err != nil {
+		if vm.Firmware, err = util.GetFirmwareFromYaml(vmConf); err != nil {
 			err = liberr.Wrap(err)
 			return
 		}
+		r.Log.Info("Setting the vm firmware ", vm.Firmware, "vmId", vm.ID)
 	case api.VSphere:
```

Comment:
@ahadas @liranr23 Shouldn't this be applicable for OVA as well?

Reply:
yep, from testing with kubevirt this filed exist so adding for OVA as well.

Reply:
then it will be nice to have also the mapping and setting the OS for OVA, as we don't do that now from what i remember